### PR TITLE
Inline Help: extract Popover content to separate component 

### PR DESCRIPTION
**Pull request by jsnajdr**
This is one of the crucial parts of #55189: extracting the Inline Help popover content from the actual `Popover` component. This content component will be rendered, without the parent `Popover` component, in the Inline Help widget.

**How to test and review:**
Verify that all needed props are passed between the `Popover` and `Content` components. Verify that the split of the SCSS stylesheet into two is done correctly. In `style.scss`, there remain only styles for the round question mark button and for the popover positioning, and everything about the content is moved the the `popover-content.scss` module.

There are two commits that remove unused CSS styles: for switching between Classic and Block editor, and for a notification dot on the help button that used to be used when Site Checklist was integrated with the Inline Help widget.

### CODE REVIEW

**Comment by jsnajdr** [comment]
There are e2e failures when clicking on the "Show me where..." search results which is weird because they're working for me when I'm clicking on them as a human. Investigating...

**Comment by jsnajdr** [comment]
So, the e2e tests were failing not in the Inline Help popover itself, but on the Home page when doing a help search there:

Screenshot 2022-01-03 at 12 28 49

And the reason was that the PR moved some of the `.inline-help__results` and `inline-help__search` styles to the `popover-content.scss` file, and these styles were not loaded on the Home page. They started to be async-loaded only after actually opening the popover.

And the e2e test depends on some of these styles being applied and active. I didn't look into the exact reasons, but it's most likely something with checking whether an element is visible or not, which is a CSS-influenced thing.

The solution is to move some styles from `popover-content.scss` to component-specific stylesheets that are imported by the affected component itself.

After this fix all checks are green again and the PR is ready for review :+1:

File: client/blocks/inline-help/inline-help-compact-results.jsx
Comment (tyxla) [inline client/blocks/inline-help/inline-help-compact-results.jsx:9]: It's not ideal that those live in 2 places. Ideally, they should be managed by a single component, and whether we'll have a "inline-help__results-cell" wrapper (and the other differences) would be determined by props.

File: client/blocks/inline-help/inline-help-search-card.jsx
Comment (tyxla) [inline client/blocks/inline-help/inline-help-search-card.jsx:9]: The `inline-help__search` classname under which all CSS inside this stylesheet is nested is not being used directly in this component, so at first glance it feels odd to import this stylesheet here.

File: client/blocks/inline-help/inline-help-search-results.scss
Comment (tyxla) [inline client/blocks/inline-help/inline-help-search-results.scss]: I think we can get rid of this one.

File: client/blocks/inline-help/inline-help-search-results.scss
Comment (tyxla) [inline client/blocks/inline-help/inline-help-search-results.scss:58]: It's a bummer we're using spans and not a classnames for the title wrapper. Pre-existing, so feel free to ignore.

File: client/blocks/inline-help/popover-content.scss
Comment (tyxla) [inline client/blocks/inline-help/popover-content.scss:69]: At first glance, it appears that this classname is not used anywhere, but at a second glance it appears to be used here:

https://github.com/Automattic/wp-calypso/blob/a82f8469a286519ffe3e68793b5d61778cf34e07/client/blocks/inline-help/popover-content.jsx#L147

File: client/blocks/inline-help/style.scss
Comment (tyxla) [inline client/blocks/inline-help/style.scss:10]: Confirmed to be unused 👍

File: client/blocks/inline-help/style.scss
Comment (tyxla) [inline client/blocks/inline-help/style.scss:72]: Confirmed to be unused. Seems like we can also remove it from here:

https://github.com/Automattic/wp-calypso/blob/7fcaec0432d9d76c548f2fe998025d691a7f79ea/client/assets/stylesheets/directly.scss#L160

File: client/blocks/inline-help/popover-content.scss
Comment (tyxla) [inline client/blocks/inline-help/popover-content.scss:4]: Noting that these `p` and `a, button` styles were previously applied to `.inline-help__mobile-overlay`. Should we preserve that?

File: client/blocks/inline-help/popover-content.scss
Comment (tyxla) [inline client/blocks/inline-help/popover-content.scss]: This lives only in the `InlineHelpForumView` component. Should we move it to its own stylesheet there?

File: client/blocks/inline-help/popover-content.scss
Comment (tyxla) [inline client/blocks/inline-help/popover-content.scss:58]: This one doesn't seem to be used right now. Actually, it seems like it was never used since it was introduced in #24390.

File: client/my-sites/customer-home/cards/features/help-search/style.scss
Comment (tyxla) [inline client/my-sites/customer-home/cards/features/help-search/style.scss:162]: We'll need to test all affected flows thoroughly. Stylesheet load order is known to be different between development and production, so there might be subtle unexpected differences, caused by different load order of stylesheets.

**Review by tyxla — COMMENTED** [review]
Thanks for splitting the giant stylesheet, this is great work!

I've left some comments, observations and suggestions after taking a more thorough look.

Also, I'd really appreciate it if we can have thorough testing instructions in the PR, because we'll need them, especially when merging, in order to ensure that stylesheet loading order doesn't create problems in production.

File: client/blocks/inline-help/inline-help-search-card.jsx
Comment (jsnajdr) [reply client/blocks/inline-help/inline-help-search-card.jsx:9]: Yes, it's used as:
```jsx
<div className="inline-help__search">
  <HelpSearchCard />
  <HelpSearchResults />
</div>
```
It doesn't have its own component where the styles could live, but always travels together with the `HelpSearchCard` component. So including the styles there is a dirty trick, but it works. Can be cleanup as one of follow-ups.

File: client/my-sites/customer-home/cards/features/help-search/style.scss
Comment (jsnajdr) [reply client/my-sites/customer-home/cards/features/help-search/style.scss:162]: I added these `.card` classes when I noticed that the search results styles are sometimes broken. See the padding between the search input and the results on the screenshot in this comment: https://github.com/Automattic/wp-calypso/pull/59413#issuecomment-1004037094 The padding is too big, due to style order ambiguity. Agreed that it needs more testing.

File: client/blocks/inline-help/popover-content.scss
Comment (jsnajdr) [reply client/blocks/inline-help/popover-content.scss:58]: This is used, just like the `inline-help__contact` class. It's one of the secondary views styled as `inline-help__${ this.state.activeSecondaryView }`

File: client/blocks/inline-help/popover-content.scss
Comment (jsnajdr) [reply client/blocks/inline-help/popover-content.scss]: Moved in e48c4fbe49

File: client/blocks/inline-help/popover-content.scss
Comment (jsnajdr) [reply client/blocks/inline-help/popover-content.scss:4]: No, they were applied to `.inline-help__popover.popover`. But the GitHub diff viewer collapses the unchanged parts of `style.scss` so that it appears they were inside `.inline-help__mobile-overlay`. Please have one more look.

File: client/blocks/inline-help/style.scss
Comment (jsnajdr) [reply client/blocks/inline-help/style.scss:72]: That seems to be a very different place, inside the Directly widget's markup. There's a `.thread-view-title-container.has-notification` class, displaying notifications on a discussion thread?

File: client/blocks/inline-help/inline-help-search-results.scss
Comment (jsnajdr) [reply client/blocks/inline-help/inline-help-search-results.scss]: Removed the commented-out styles in 15c1a008e8

**Comment by jsnajdr** [comment]
**Testing instruction:**
Verify all the places where help search and results are displayed. There are plenty of them!

In the Inline Help popover, there's the default view with help search:

Screenshot 2022-01-05 at 13 48 59

After clicking "Contact us" and typing a message in the chat box, you'll see some contextual results before your message is really sent:

Screenshot 2022-01-05 at 13 48 25

Then there are the `/help` pages. On the `/help` page, there is help search, which shows some suggestions (rendered using the "help results" components) even before you start searching:

Screenshot 2022-01-05 at 13 51 48

After you start searching, some "real" results are shown:

Screenshot 2022-01-05 at 13 53 16

Then after clicking "Contact support" you'll be taken to `/help/contact`, and while typing into the chat message box, you'll see some contextual results again:

Screenshot 2022-01-05 at 13 55 54

Finally there's the Home screen where you can also search help:

Screenshot 2022-01-05 at 13 57 24

All these places share some common styles (like `.help-search .card.search-card`) and can be affected when the styles are modified.

**Comment by jsnajdr** [comment]
Howdy @tyxla :wave: I addressed all the review comments, and also posted some detailed testing instructions. I believe the PR is ready for another round of review.

**Review by tyxla — APPROVED** [review]
Code looks good! 

I've also done some thorough testing and it works great!

It could use a rebase before landing (I noticed that the bug you fixed in the footer in #59668 was still not in this branch).

Nice work 💯

**Comment by jsnajdr** [comment]
Rebased this onto the latest `trunk` (the base commit was indeed quite old) and will merge once all checks come back green :+1:

**Pull request merged**
